src/commands/Minion/mix.ts

```diff
@@ -0,0 +1,181 @@
+import { Activity, Tasks, Time } from '../../lib/constants';
+import { minionNotBusy, requiresMinion } from '../../lib/minions/decorators';
+import { UserSettings } from '../../lib/settings/types/UserSettings';
+import Herblore from '../../lib/skilling/skills/herblore/herblore';
+import { SkillsEnum } from '../../lib/skilling/types';
+import { HerbloreActivityTaskOptions } from '../../lib/types/minions';
+import {
+	bankHasItem,
+	formatDuration,
+	itemNameFromID,
+	removeItemFromBank,
+	stringMatches
+} from '../../lib/util';
+import addSubTaskToActivityTask from '../../lib/util/addSubTaskToActivityTask';
+
+export default class extends BotCommand {
+	public constructor(store: CommandStore, file: string[], directory: string) {
+		super(store, file, directory, {
+			aliases: ['crush', 'clean'],
+			altProtection: true,
+			oneAtTime: true,
+			cooldown: 1,
+			usage: '[quantity:int{1}|name:...string] [name:...string]',
+			usageDelim: ' '
+		});
+	}
+
+	@minionNotBusy
+	@requiresMinion
+	async run(msg: KlasaMessage, [quantity, mixName = '']: [null | number | string, string]) {
+		if (msg.flagArgs.items) {
+			return msg.channel.sendFile(
+				Buffer.from(
+					Herblore.Mixables.map(
+						item =>
+							`${item.name} - lvl ${item.level} : ${Object.entries(item.inputItems)
+								.map(entry => `${entry[1]} ${itemNameFromID(parseInt(entry[0]))}`)
+								.join(', ')}`
+					).join('\n')
+				),
+				`Available Herblore potions and items.txt`
+			);
+		}
+
+		if (typeof quantity === 'string') {
+			mixName = quantity;
+			quantity = null;
+		}
+		const mixableItem = Herblore.Mixables.find(item =>
+			item.aliases.some(alias => stringMatches(alias, mixName))
+		);
+
+		if (!mixableItem) {
+			return msg.send(
+				`That is not a valid mixable item, to see the items available do \`${msg.cmdPrefix}mix --items\``
+			);
+		}
+
+		if (msg.author.skillLevel(SkillsEnum.Herblore) < mixableItem.level) {
+			return msg.send(
+				`${msg.author.minionName} needs ${mixableItem.level} Herblore to make ${mixableItem.name}.`
+			);
+		}
+
+		if (
+			mixableItem.qpRequired &&
+			msg.author.settings.get(UserSettings.QP) < mixableItem.qpRequired
+		) {
+			return msg.send(
+				`You need atleast **${mixableItem.qpRequired}** QP to make ${mixableItem.name}.`
+			);
+		}
+
+		let sets = 'x';
+		let cost = 'is now';
+		if (mixableItem.outputMultiple) {
+			sets = 'batches of';
+		}
+
+		await msg.author.settings.sync(true);
+		const userBank = msg.author.settings.get(UserSettings.Bank);
+		let requiredItems: [string, number][] = Object.entries(mixableItem.inputItems);
+
+		// Get the base time to mix the item then add on quarter of a second per item to account for banking/etc.
+		let timeToMixSingleItem =
+			mixableItem.tickRate * Time.Second * 0.6 + mixableItem.bankTimePerPotion * Time.Second;
+
+		const zahur = Boolean(msg.flagArgs.zahur);
+		if (zahur && mixableItem.zahur === true) {
+			timeToMixSingleItem = 0.000001;
```

Comment:
instead of sending this as a activity, handle the action straight away similar to how +buy, +create, +wyson and such works.